src/RandomForest.jl

```diff
@@ -76,7 +76,8 @@ function predict(forest::RandomForest, data::AbstractMatrix)
         labels[index, :] = labels_for_tree
     end
     # Calculate the mode of every sample
-    is_regression = eltype(identity.(labels)) <: Number
+    is_regression = (forest.split_criterion in get_split_criterions() && forest.split_criterion in get_split_criterions("regression")) ||
```

Comment:
I also fixed a bug regarding this line. 

Say i used split_gini as split criterion with a dataset that has integers as labels. I clearly want to do classification, but this lead to the mean being calculated which does not make sense. 

Now, if one of the implemented criterions is used it checks regarding the criterion if its classification or regression and else uses the previous check